### PR TITLE
add tests and code to return single and archive checksums

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/111525

This PR leverages the work done in moderntribe/tribe-common#747 to return checksums in Event REST API responses:

for single events the checksum is the checksum of the event itself
for archive two checksums are returned page_checksum for the checksum of the current page, and request_checksum for the checksum relative to the whole request; the two could have the same values if no pagination is involved
I've added some logic in the Tribe__Events__Query class and the last_found_ids method.

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Expected 1 space before &quot;=&gt;&quot;; 0 found | [Event_Definition_Provider.php:26](https://github.com/moderntribe/the-events-calendar/blob/ceb189d79a2ee3fb1d08a4475c5012ae9d6257cf/src/Tribe/REST/V1/Documentation/Event_Definition_Provider.php#L26)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Expected 1 space before &quot;=&gt;&quot;; 0 found | [Event_Definition_Provider.php:26](https://github.com/moderntribe/the-events-calendar/blob/b4a03361203f625dcecef0a9f8b145cf971bc388/src/Tribe/REST/V1/Documentation/Event_Definition_Provider.php#L26)

> via jenkins/codesniffer-phpcs

File: src/Tribe/REST/V1/Post_Repository.php [hunk 1]
```diff
@@ -76,6 +76,7 @@ public function get_event_data( $event_id, $context = '' ) {
 
 		$data = array(
 			'id'                     => $event_id,
+			'checksum'               => tribe_post_checksum( $event_id ),
```
Comment (sc0ttkclark) [inline src/Tribe/REST/V1/Post_Repository.php]: Why not pass in $event here to avoid additional get_post call?

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Expected 1 space before &quot;=&gt;&quot;; 0 found | [Event_Definition_Provider.php:26](https://github.com/moderntribe/the-events-calendar/blob/1ebe6315bdb62f3b276b950edcbbfa4bbb08f41d/src/Tribe/REST/V1/Documentation/Event_Definition_Provider.php#L26)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Expected 1 space before &quot;=&gt;&quot;; 0 found | [Event_Definition_Provider.php:26](https://github.com/moderntribe/the-events-calendar/blob/dc47a113071f39bfb512a763b51248cb669f0774/src/Tribe/REST/V1/Documentation/Event_Definition_Provider.php#L26)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Expected 1 space before &quot;=&gt;&quot;; 0 found | [Event_Definition_Provider.php:26](https://github.com/moderntribe/the-events-calendar/blob/c3f84633b72a14300c0921141343ec4e88020919/src/Tribe/REST/V1/Documentation/Event_Definition_Provider.php#L26)

> via jenkins/codesniffer-phpcs

File: src/Tribe/REST/V1/Post_Repository.php (hunk 1 above)
Comment (lucatume) [reply src/Tribe/REST/V1/Post_Repository.php]: Good catch, thanks.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Expected 1 space before &quot;=&gt;&quot;; 0 found | [Event_Definition_Provider.php:26](https://github.com/moderntribe/the-events-calendar/blob/e482f96bda165770da1907600757541b8cccc4c9/src/Tribe/REST/V1/Documentation/Event_Definition_Provider.php#L26)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Each line in an array declaration must end in a comma | [Event_Definition_Provider.php:121](https://github.com/moderntribe/the-events-calendar/blob/3fad3c87e3a38b166f2e94b91bb032adacb4c12b/src/Tribe/REST/V1/Documentation/Event_Definition_Provider.php#L121)
:x: | Each line in an array declaration must end in a comma | [Event_Definition_Provider.php:130](https://github.com/moderntribe/the-events-calendar/blob/3fad3c87e3a38b166f2e94b91bb032adacb4c12b/src/Tribe/REST/V1/Documentation/Event_Definition_Provider.php#L130)

> via jenkins/codesniffer-phpcs

**Comment by lucatume** [comment]
Updated and good for another look

**Pull request merged**